HUSB238 component fixes/improvements

AndreiPustovalov opened the pull request:
* fix voltage/current for non-PD compatible chargers
* fix binary sensors non reporting issue
* reduce I2C request frequency by moving read_all_ from loop() to update()
* send updated values right after new voltage selected
* publish values only if something changed

# What does this implement/fix?

<!-- Quick description and explanation of changes -->

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code quality improvements to existing code or addition of tests
- [ ] Other

**Related issue or feature (if applicable):**

- fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):**

- esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [x] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

AndreiPustovalov committed:
HUSB238 component fixes/improvements

* fix voltage/current for non-PD compatible chargers
* fix binary sensors non reporting issue
* reduce I2C request frequency by moving read_all_ from loop() to update()
* send updated values right after new voltage selected
* publish values only if something changed

AndreiPustovalov commented:
I fixed all the issues I found, it seems to be working fine, but I don't have any experience with esphome, so my solution might be not optimal.

latonita reviewed (changes requested):
would you please pass is_changed as reference instead of a pointer

AndreiPustovalov committed:
HUSB238 Component bugfixes and code style fix

* pass is_changed as a reference instead of a pointer
* don't overwrite register data in select_pdo_voltage_ as it is used later in comparison with previous state and it's not guaranteed that it will be updated
* do not call command_get_src_cap as it has no effect, but resets selected voltage
* remove unused code

AndreiPustovalov commented:
> would you please pass is_changed as reference instead of a pointer

thanks, good point. fixed.
I found couple more issues and fixed them as well.

sometimes there are warnings about component taking to long to update (~50ms), usually when there are i2c errors. Is it a problem?

latonita commented:
@AndreiPustovalov thanks. I have merged your changes , restored API function for the use from yaml.
Please test the latest version.

AndreiPustovalov commented:
@latonita, I tested the latest version and wrote comment about it in PR to esphome [here](https://github.com/esphome/esphome/pull/6693#issuecomment-2561230978)

everything works, thank you!
